gensim/models/poincare.py

```diff
@@ -167,47 +171,48 @@ def __init__(self, train_data, size=50, alpha=0.1, negative=10, workers=1, epsil
         self._np_random = np_random.RandomState(seed)
         self.init_range = init_range
         self._loss_grad = None
-        self._load_relations()
-        self._init_embeddings()
+        self.build_vocab(train_data)
 
-    def _load_relations(self):
+    def build_vocab(self, relations=None, update=False):
         """Load relations from the train data and build vocab."""
-        vocab = {}
-        index2word = []
-        all_relations = []  # List of all relation pairs
-        node_relations = defaultdict(set)  # Mapping from node index to its related node indices
+        old_index2word_len = len(self.kv.index2word)
 
         logger.info("loading relations from train data..")
-        for relation in self.train_data:
+        for relation in relations:
             if len(relation) != 2:
                 raise ValueError('Relation pair "%s" should have exactly two items' % repr(relation))
             for item in relation:
-                if item in vocab:
-                    vocab[item].count += 1
+                if item in self.kv.vocab:
```

Comment:
Ask for forgiveness instead of permission - it's more Pythonic.

```python
try:
   dict[item] += 1
except KeyError:
   dict[item] = 0
```

https://stackoverflow.com/questions/12265451/ask-forgiveness-not-permission-explain

Reply:
-1: The original code (current version) looks clearer and better to me. Obfuscating the logic with unnecessary exception handling would be a regression IMO.

Reply:
OK, up to you.